lib/src/bump_version.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:sidekick_core/sidekick_core.dart';
+
+extension BumpVersion on Version {
+  /// Returns a bumped [Version], major, minor or patch
+  ///
+  /// Always bumps buildNumber when set by 1
+  Version bumpVersion(VersionBumpType bumpType) {
+    final oldBuildNumber = build.firstOrNull as int?;
+    Version newVersion = this;
+    switch (bumpType) {
+      case VersionBumpType.major:
+        newVersion = nextMajor;
+        break;
+      case VersionBumpType.minor:
+        newVersion = nextMinor;
+        break;
+      case VersionBumpType.patch:
+        newVersion = nextPatch;
+        break;
+    }
+
+    if (oldBuildNumber != null) {
+      final newBuildNumber = oldBuildNumber + 1;
+      // build is immutable and null if not present
+      newVersion = newVersion.copyWith(build: newBuildNumber.toString());
```

Comment:
This throws away all other parts of a build number, e.g.

```dart
final v = Version(1, 2, 3, build: '4.5.6');
print(v); // prints 1.2.3+4.5.6
print(v.bumpVersion(VersionBumpType.major)); // prints 2.0.0+5
```

I think it would be safer to bump the build number only when we can safely do so (when there is only one `int` element in `build`):

```dart
final newBuild = build.whereType<int>().length == 1
    ? build.map((e) => e is int ? e++ : e)
    : null;
```

Reply:
Good idea, let's do this behavior change in another PR :)